lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -321,17 +359,45 @@ def headers
         }
       end
 
-      def url(_post, action, authorization)
-        if %i[authorize purchase credit].include?(action)
+      def public_key_headers
+        {
+          'Authorization' => @options[:public_key],
+          'Content-Type' => 'application/json;charset=UTF-8'
+        }
+      end
```

Comment:
Seems like this is redundant and we can put a conditional in the `headers` method to use `public_key` if we are doing a `store` transaction.